### PR TITLE
feat(dws): add datasource to query workload plans

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

Add new datasource to query list of the workload plans under specified DWS cluster.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new datasource huaweicloud_dws_workload_plans.
2. add related document and acceptance test.

```

## PR Checklist

* [x] Tests added/passed.

```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccSomethingV0_basic'
...
 make testacc TEST=./huaweicloud/services/acceptance/dws TESTARGS='-run TestAccDataSourceWorkloadPlans_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccDataSourceWorkloadPlans_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceWorkloadPlans_basic
=== PAUSE TestAccDataSourceWorkloadPlans_basic
=== CONT  TestAccDataSourceWorkloadPlans_basic
--- PASS: TestAccDataSourceWorkloadPlans_basic (347.68s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       347.732s
```

* [x] Documentation updated.
* [x] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/dws TESTARGS='-run TestAccDataSourceWorkloadPlans_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccDataSourceWorkloadPlans_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceWorkloadPlans_basic
=== PAUSE TestAccDataSourceWorkloadPlans_basic
=== CONT  TestAccDataSourceWorkloadPlans_basic
--- PASS: TestAccDataSourceWorkloadPlans_basic (281.04s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       281.105s

File: docs/data-sources/dws_workload_plans.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  The [plans](#plans_struct) structure is documented below.
+
+<a name="plans_struct"></a>
+The `plans` block supports:
+
+* `id` - The ID of the workload plan.
+
+* `name` - The name of the workload plan.
+
+* `cluster_id` - The cluster ID of to which the workload plan belongs.
```
Comment (ruwenqiang123) [inline docs/data-sources/dws_workload_plans.md]:
```suggestion
* `cluster_id` - The cluster ID to which the workload plan belongs.
```

File: docs/data-sources/dws_workload_plans.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the resource.
+  If omitted, the provider-level region will be used.
+
+* `cluster_id` - (Required, String) Specifies the DWS cluster ID to which the workload plans belong.
+
+* `logical_cluster_name` - (Optional, String) Specifies the logical cluster name to which the workload plans belong.
+  This parameter is only available when The DWS cluster is logical cluster.
```
Comment (ruwenqiang123) [inline docs/data-sources/dws_workload_plans.md]:
```suggestion
  This parameter is only available when the DWS cluster is logical cluster.  
```

**Pull request merged**